blocks/bricks/sequence_generators.py

```diff
@@ -162,7 +162,7 @@ def cost(self, application_call, outputs, mask=None, **kwargs):
 
         # Separate the deliverables
         states = {name: results[name][:-1] for name in self.state_names}
-        glimpses = {name: results[name] for name in self.glimpse_names}
+        glimpses = {name: results[name][1:] for name in self.glimpse_names}
```

Comment:
+1 for more comments in this file; it's not the easiest code to read because of how tightly integrated all the bricks here are